### ISSUE TITLE
[FEAT] Persist DHT peer data using e.g. Badger

### ISSUE THREAD

**Issue by SebastianJ** [ElrondNetwork/elrond-go#1948]
Given the recent boot/seed node issues - it's probably a good idea to look into persisting the DHT peer state using e.g. [Badger](https://github.com/dgraph-io/badger)

Here's a simple example for how it can be done:

```
package p2p

import (
	"context"

	badger "github.com/ipfs/go-ds-badger"
	libp2p_host "github.com/libp2p/go-libp2p-core/host"
	libp2p_dht "github.com/libp2p/go-libp2p-kad-dht"
	libp2p_dhtopts "github.com/libp2p/go-libp2p-kad-dht/opts"
)

func dhtWithPersistance(dbPath string, host libp2p_host.Host) (*libp2p_dht.IpfsDHT, error) {
	ctx, _ := context.WithCancel(context.Background())

	var dhtOpts []libp2p_dhtopts.Option

	if dbPath != "" {
		db, err := badger.NewDatastore(dbPath, nil)
		if err != nil {
			return nil, err
		}

		dhtOpts = append(dhtOpts, libp2p_dhtopts.Datastore(db))
	}

	dht, err := libp2p_dht.New(ctx, host, dhtOpts...)
	if err != nil {
		return nil, err
	}

	return dht, nil
}
```

(I already posted about this on Telegram - but thought it might be useful to report it here as well)

**Comment by iulianpascalau** [comment]
So I came with updates: @bogdan-rosianu was on this issue and we tried to use a wrapper over our storage implementations and during testing we found out that the kad-dht implementation does not actually call put. It relies on the peerstore implementation fetched from the node. So I think that should be implemented with a backup storage. Since the peerstore is a bit of a monster now, I think we will postpone this a little bit. I think the example you provided is suited if we actually have used the DHT mechanism for storing data in a distributed manner (just like ipfs does with the stored files). Thanks for the ideea & feedback :+1: . Will remain open for the time being.
